File: framework/src/main/java/com/djangoogle/framework/util/NoDoubleClickUtils.java
package com.djangoogle.framework.util;

/**
 * 防重复点击
 * Created by Djangoogle on 2018/10/11 10:31 with Android Studio.
 */
public class NoDoubleClickUtils {

	public static final long INTERVAL = 500L;
	private static long lastClickTime;

	public synchronized static boolean isDoubleClick() {
		long currentTime = System.currentTimeMillis();
		boolean isClick = currentTime - lastClickTime < INTERVAL;
		lastClickTime = currentTime;
		return isClick;
	}
}

File: framework/src/main/java/com/djangoogle/framework/util/NoDoubleClickUtils.kt
package com.djangoogle.framework.util

/**
 * 防重复点击
 * Created by Djangoogle on 2018/10/11 10:31 with Android Studio.
 */
class NoDoubleClickUtils {

	companion object {

		private var lastClickTime: Long = 0L

		@Synchronized
		fun isDoubleClick(interval: Long): Boolean {
			val currentTime = System.currentTimeMillis()
			val isClick = currentTime - lastClickTime < interval
			lastClickTime = currentTime
			return isClick
		}
	}
}
